hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -159,12 +163,114 @@ public long getBytesReceived() {
     return bytesReceived;
   }
 
-  public ListResultSchema getListResultSchema() {
-    return listResultSchema;
+
+  protected void setStatusCode(final int statusCode) {
```

Comment:
Should all these properties reside in the base class?

Are they used in the AbfsFastpathConnection?
If not, I think the property should be bubbled up into the AbfsHttpConnection class.

Reply:
Yes, even when on Fastpath, driver will receive HttpStatus code and other relevant details for the corresponding request, this enables existing logic for retry/throttling handling to continue to work. Has to be on base class.